### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by mykyta-protsenko**

### CODE REVIEW

**Comment by mykyta-protsenko** [comment]
@anavianm I've noticed that checks have failed - can you double-check to make sure no commits are missing?

A force-push to master may have screwed up a git history. On top of that it would be good to run `python manage.py test` locally.

```
  File "/opt/hostedtoolcache/Python/3.7.7/x64/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 724, in exec_module
  File "<frozen importlib._bootstrap_external>", line 860, in get_code
  File "<frozen importlib._bootstrap_external>", line 791, in source_to_code
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/runner/work/Simpact/Simpact/simpact/urls.py", line 24
    path('simpact/', include('SimpactApp.urls'))
       ^
SyntaxError: invalid syntax
```

File: users/models.py [hunk 1]
```diff
@@ -1,3 +1,12 @@
+from django.contrib.auth.models import AbstractUser
 from django.db import models
 
+class User(AbstractUser):
+    is_volunteer = models.BooleanField(default=False)
+    is_ngo = models.BooleanField(default=False)
+    is_business = models.BooleanField(default=False)
+
+class Volunteer(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE, primary_key=True)
```
Comment (mykyta-protsenko) [inline users/models.py:12]: I guess Volunteer class will have more volunteer-related tables? Might be worth checking @baffour1999 's PR.

File: users/views/volunteer.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+@method_decorator([login_required, volunteer_required], name = 'dispatch')
+class VolunteerDashboardView(UpdateView):
+    model = Volunteer
+    form_class = VolunteerDashbaordForm
+    template_name = 'templates/users/VDashboard.html'
+
+@login_required
+@volunteer_required
+def volunteer_dashbaord(request, pk):
```
Comment (mykyta-protsenko) [inline users/views/volunteer.py]: `volunteer_dashbaord` looks like a typo :)

File: users/models.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ class perksForm(models.Model):
 
 
 class advertisementForm(models.Model):
-    owner = models.ForeignKey(User, on_delete = models.CASCADE, related_name = 'perksSubmit')
```
Comment (anavianm) [inline users/models.py]: thank you I didnt catch ths!

**Review by anavianm — APPROVED** [review]

**Review by baffour1999 — APPROVED** [review]

**Pull request closed without merging**